File: 2022/02/py/1.py
with open("i.txt") as f:
    data = [l.removesuffix("\n") for l in f]

n = 0
for l in data:
    if not l:
        continue
    a=0
    b = 0
    if l[0] == "A":
        a = 0
    elif l[0] == "B":
        a = 1
    elif l[0] == "C":
        a = 2
    if l[2] == "X":
        b = 0
    elif l[2] == "Y":
        b = 1
    elif l[2] == "Z":
        b = 2

    if b-a == 1 or b-a == -2:
        n += 6
    elif b-a == 0:
        n += 3
    n+= b+1

print( n )

File: 2022/02/js/1.js
const fs = require("fs");
let data = fs.readFileSync("i.txt").toString();
console.log(data.split(/\r?\n/)
    .map(e => [e.charCodeAt(0), e.charCodeAt(2)])
    .map(([a, b]) => (b - 81 - ((a - b + 27) % 3) * 3))
    .reduce((acc, n) => acc + n));
